taskcluster/known_problems.rst

```diff
@@ -62,6 +62,45 @@ Scan the `ci-configuration`_ repo for your project and see if we were granting
 any special scopes to the old branch. If so, update the name to the new
 branch and land.
 
+.. _missing mercurial features:
+
+Missing Mercurial features when Cloning the Repo
+------------------------------------------------
+
+Symptoms
+~~~~~~~~
+
+After updating the version of Mercurial used on newer branches, tasks on older
+branches might start failing with errors like:
+
+.. parsed-literal::
+   abort: repository requires features unknown to this Mercurial: revlog-compression-zstd!
+
+This happens because repos cloned via newer versions of Mercurial are often
+incompatible with older versions of Mercurial. Since workers have a shared
+checkout cache between tasks, if a worker first claims a task on e.g,
+`mozilla-central`, clones the repo, and then claims a task on
+`mozilla-release`, the latter task might fail if it is using an older
+Mercurial.
+
+Solution
+~~~~~~~~
+
+The solution is to increment the cache identifier associated with the checkout
+cache. For example, the Gecko decision tasks use `this cache name`_. Changing
+the name (e.g incrementing `v2` -> `v3`), will ensure workers don't re-use the
+same cache across disparate branches.
+
+The downside to doing this is that tasks on older branches with less traffic
+will start having many more cache misses, resulting in longer runtimes which
+can impact our ability to ship expediently.
+
+So in addition to bumping the cache name, the image bump that caused Mercurial
```

Comment:
I'm curious about this.  There's not a ton of decision tasks on release (very few pushes, and one or two a day from cron, maybe?).  beta has more, and esr is long-lived, so there might be a benefit there, OTOH once a worker has run one decision task from one of those branches it'll have populated the cache, so I'm not completely convinced it's that big a deal.